addons/l10n_hk/models/res_bank.py

```diff
@@ -10,50 +10,64 @@
+                                  ondelete={'id': 'set default', 'mobile': 'set default', 'email': 'set default'})
 
-    @api.constrains('l10n_hk_fps_type', 'l10n_hk_fps_identifier')
-    def _check_l10n_hk_fps_identifier(self):
+    @api.constrains('proxy_type', 'proxy_value', 'partner_id')
+    def _check_proxy(self):
         auto_mobn_re = re.compile(r"^[+]\d{1,3}-\d{6,12}$")
         for bank in self:
             if bank.country_code != 'HK':
                 continue
-            if bank.l10n_hk_fps_type == 'id' and (not bank.l10n_hk_fps_identifier or len(bank.l10n_hk_fps_identifier) not in [7, 9]):
+            if bank.proxy_type == 'id' and (not bank.proxy_value or len(bank.proxy_value) not in [7, 9]):
                 raise ValidationError(_("Invalid FPS ID! Please enter a valid FPS ID with length 7 or 9."))
-            if bank.l10n_hk_fps_type == 'mobile' and (not bank.l10n_hk_fps_identifier or not auto_mobn_re.match(bank.l10n_hk_fps_identifier)):
+            if bank.proxy_type == 'mobile' and (not bank.proxy_value or not auto_mobn_re.match(bank.proxy_value)):
                 raise ValidationError(_("Invalid Mobile! Please enter a valid mobile number with format +852-67891234."))
-            if bank.l10n_hk_fps_type == 'email' and (not bank.l10n_hk_fps_identifier or not single_email_re.match(bank.l10n_hk_fps_identifier)):
+            if bank.proxy_type == 'email' and (not bank.proxy_value or not single_email_re.match(bank.proxy_value)):
                 raise ValidationError(_("Invalid Email! Please enter a valid email address."))
 
+    @api.depends('country_code')
+    def _compute_display_qr_setting(self):
+        bank_hk = self.filtered(lambda b: b.country_code == 'HK')
+        bank_hk.display_qr_setting = True
+        super(ResPartnerBank, self - bank_hk)._compute_display_qr_setting()
+
     # Follow the documentation of FPS QR Code Standard [1]
     # [1]: https://www.hkma.gov.hk/media/eng/doc/key-functions/financial-infrastructure/infrastructure/retail-payment-initiatives/Common_QR_Code_Specification.pdf
     def _get_merchant_account_info(self):
         if self.country_code == 'HK':
             fps_type_mapping = {
-                'id': '02',
-                'mobile': '03',
-                'email': '04'
+                'id': 2,
+                'mobile': 3,
+                'email': 4,
             }
-            fps_type_id = fps_type_mapping[self.l10n_hk_fps_type]
-            fps_account = f'{fps_type_id}{len(self.l10n_hk_fps_identifier):02}{self.l10n_hk_fps_identifier}'
-            return f'0012hk.com.hkicl{fps_account}'
+            fps_type = fps_type_mapping[self.proxy_type]
+            merchant_account_vals = [
+                (0, 'hk.com.hkicl'),                                            # GUID
+                (fps_type, self.proxy_value),                        # Proxy Type and Proxy Value
```

Comment:
```suggestion
                (0, 'hk.com.hkicl'),                                 # GUID
                (fps_type, self.proxy_value),                        # Proxy Type and Proxy Value
```
May as well keep it aligned :smile: